### ISSUE TITLE
Ugly formatting with `extends`

### ISSUE THREAD

**Issue by so1ve** [prettier/prettier#15259]
**Prettier 3.0.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACAS1jmwDMBDMOPACRhgwCU4BnLKFuAHgB0o88AETzp4UACYs8AVygBrKBADuUADR8BAURFoxkmfMUq8AXgMLlUPgD4deqczbROeYHx4xx5GOSRCA3O7w2Ng4fpqBUAC+IKogEBgwBM7IoOQhygAK6QgsyCDkADZK5ACeeXEARtiUcnAwAMrkALZwADJEcMgUhZxVNWB1jRiURADmyDDY0nBxnM0Ek9OzICzjhXAAitIQ8N1FfSAAVixoDetbO3tIPYcAjldwmaEYeSjkLAC0UHBw4n+xEBTcgEQrjADCEGazXI+SKhUBaygYw2AEF6NgCJVpPBMiQOj99r0VgALGDNQoAdRJBHgbEocAauVpBAAbrTSvkwCwKiBWTMAJISBCNMCYxKoiQNGClDZEw4YUKcSk1DD5RWsEisrpxIicXDPchjGHylYjbD6-IyjCsMUERKAxVEGCUgjiGAk5AADgADHFsHAHgQA4bjbCbgcVj5Kq73Z6kAAmOLSTgAFXIlTetxWcGalT+-3EbXIyOkRrgADEcDD6OM4TiICAolEgA)

```sh
--parser typescript
```

**Input:**

```tsx
export interface HttpResponse<
  D extends unknown,
  E extends unknown = unknown
> extends Response {
	data: D;
	error: E;
}
```

**Output:**

```tsx
export interface HttpResponse<D extends unknown, E extends unknown = unknown>
  extends Response {
  data: D;
  error: E;
}

```

**Expected behavior:**

IMO the input is better, prettier should not place `extends` in a newline

**Comment by sosukesuzuki** [comment]
Duplicated https://github.com/prettier/prettier/issues/15219

**Comment by so1ve** [comment]
Hi @sosukesuzuki, thanks for quick response. But are you sure they are identical? Seemed #15219 is about braces.

**Comment by sosukesuzuki** [comment]
Ah sorry I misunderstood

**Comment by so1ve** [comment]
And this one:

```ts
export interface DenoLambdaOptions
	extends Omit<LambdaOptionsWithFiles, 'runtime' | 'supportsWrapper'> {}
```

**Comment by seiyab** [comment]
@so1ve 
What is desired output for https://github.com/prettier/prettier/issues/15259#issuecomment-1683980481?
This one?
```ts
export interface DenoLambdaOptions extends Omit<
	LambdaOptionsWithFiles,
	"runtime" | "supportsWrapper"
> {}
```

IMHO, https://github.com/prettier/prettier/issues/15259#issuecomment-1683980481 looks as fine as this one↑.

**Comment by so1ve** [comment]
Yes I agree https://github.com/prettier/prettier/issues/15259#issuecomment-1683980481 doesn't seemed to be ugly 👍
